cfme/tests/configure/test_proxy.py

```diff
@@ -29,27 +31,48 @@ def proxy_machine():
     After test run vm deletes from provider
     """
     depot_machine_name = random_vm_name('proxy')
-    data = conf.cfme_data.get("proxy_template")
+    data = conf.cfme_data.get("utility_vm")
     proxy_provider_key = data["provider"]
     proxy_template_name = data["template_name"]
-    proxy_port = data['port']
+    proxy_port = '3128'
     prov = get_mgmt(proxy_provider_key)
     vm = deploy_template(proxy_provider_key,
                          depot_machine_name,
                          template_name=proxy_template_name)
     wait_for(func=lambda: vm.ip is not None, num_sec=300, delay=10,
              message='Waiting for instance "{}" ip to be present.'.format(vm.name))
 
+    credentials_key = conf.cfme_data['utility_vm']['credentials']
+    with SSHClient(
+            hostname=vm.ip,
+            **conf.credentials[credentials_key]) as ssh_client:
+        repofile = os.path.join(os.path.dirname(__file__), 'rhel.repo')
```

Comment:
There's an assumption here that the deployed template is RHEL, and I don't think that's a safe assumption to make. Its pulled from yaml, and could be anything.

I think that comments should be made here to state the assumptions that we end up making, and possibly checks for an expected proxy type/port availability on the utility vm. I do like the change to making it more of an abstract name, not specifically only for a proxy template. But I do think that here we should assume that the proxy is already installed and running on the utility VM when it starts up instead of trying to install it here.

Otherwise, we need to either attempt to pull and build squid from source so that we don't have to care about what OS we're on (assuming its one that we can build on, or we have to install whatever is needed to build - again a problem if we don't know what OS the template is).  This is the same challenge we face in quickstart - handling install of required packages on multiple OS types.

Reply:
I think this is a **Required** comment.